test/unit/grammar/type_traits.cpp

```diff
@@ -9,3 +9,32 @@
 
 // Test that header file is self-contained.
 #include <boost/url/grammar/type_traits.hpp>
+
+#include <string>
+
+#include "test_suite.hpp"
+
+namespace boost {
+namespace urls {
+namespace grammar {
+
+struct type_traits_test
+{
+    void
+    run()
+    {
+#ifdef __cpp_lib_concepts
+        BOOST_TEST(std::input_iterator<detail::input_it_ex>);
+#endif
+        BOOST_TEST(is_mutable_string<std::string>::value);
+        BOOST_TEST_NOT(is_mutable_string<int>::value);
```

Comment:
c'mon man give it something to chew on, maybe `vector<char>`? or how about a `std::string const`? Or how about `std::basic_string<char const, std::char_traits<char>>` ?